cmake/OpenCVDetectPython.cmake

```diff
@@ -88,6 +88,9 @@ function(find_python preferred_version min_version library_env include_dir_env
         find_host_package(PythonLibs "${_version_major_minor}")
       else()
         find_host_package(PythonLibs "${_version_major_minor}.${_version_patch}" EXACT)
+        if(NOT PYTHONLIBS_FOUND)
```

Comment:
Interesting, it fixes PYTHON3 detection but breaks PYTHON2 detection for me.
I think problem is several lines upper, the environment detection works wrong, it should be:
```.cmake
      ocv_check_environment_variables(${library_env} ${include_dir_env})
      if(${${library_env}})
          set(PYTHON_LIBRARY "${${library_env}}")
      endif()
      if(${${include_dir_env}})
          set(PYTHON_INCLUDE_DIR "${${include_dir_env}}")
      endif()
```